File: src/main/java/com/architecture_map/belarus/entity/Address.java
package com.architecture_map.belarus.entity;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@Getter
@Setter(AccessLevel.PUBLIC)
@RequiredArgsConstructor
@ToString
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String locality;
    private String district;
    private String region;
    private String street;
    private String houseNumber;
}

File: src/main/kotlin/by/architecturemap/belarus/entity/Address.kt
package by.architecturemap.belarus.entity

import jakarta.persistence.Entity
import jakarta.validation.constraints.NotBlank

@Entity
data class Address(

    var locality: String? = null,
    var district: String? = null,

    @NotBlank(message = "Region may not be blank")
    var region: String,

    var street: String? = null,
    var houseNumber: String? = null

) : BaseEntity()
